FirebaseMLModelDownloader/Sources/ModelDownloader.swift

```diff
@@ -436,6 +431,33 @@ extension ModelDownloader {
                 self.mainQueueHandler(completion(.failure(error)))
               }
             }
+            /// Stop keeping track of current download task.
+            self.currentDownloadTask.removeValue(forKey: modelName)
+          }
+          /// Merge duplicate requests if there is already a download in progress for the same model.
+          if let downloadTask = self.currentDownloadTask[modelName],
```

Comment:
Mutating the dictionary is not a thread-safe operation. As the method may be called from any thread by the client we need to ensure the thread-safety e.g. by calling it from a serial dispatch queue.

Reply:
Good catch, totally missed this. Please let me know if I'm doing this right.